src/Mobs/Monster.cpp

```diff
@@ -122,67 +120,68 @@ void cMonster::SpawnOn(cClientHandle & a_Client)
 
 
 
-void cMonster::TickPathFinding(cChunk & a_Chunk)
+bool cMonster::TickPathFinding(cChunk & a_Chunk)
 {
-
-	if (m_Path == nullptr)
+	if (!m_IsFollowingPath)
 	{
-		Vector3d position = GetPosition();
-		Vector3d Dest = m_FinalDestination;
-
-		// Can someone explain why are these two NOT THE SAME???
-		// m_Path = new cPath(GetWorld(), GetPosition(), m_FinalDestination, 30);
-		m_Path = new cPath(a_Chunk, Vector3d(floor(position.x), floor(position.y), floor(position.z)), Vector3d(floor(Dest.x), floor(Dest.y), floor(Dest.z)), 20);
-
+		return false;
+	}
 
-		m_IsFollowingPath = false;
+	if (ReachedFinalDestination())
+	{
+		StopMovingToPosition();
+		return false;
 	}
-	m_PathStatus = m_Path->Step(a_Chunk);
-	switch (m_PathStatus)
+
+	if (m_Path == nullptr)
 	{
+		m_Path = new cPath(a_Chunk, GetPosition().Floor(), m_FinalDestination.Floor(), 200);
+	}
 
+	switch (m_Path->Step(a_Chunk))
+	{
 		case ePathFinderStatus::PATH_NOT_FOUND:
 		{
-			ResetPathFinding();
+			StopMovingToPosition();  // Try to calculate a path again
```

Comment:
On a second thought, this is the right approach, it's just the comment that's misleading. It should be "Give up PathFinding to that destination".